src/account_x/x_account_controller.ts

```diff
@@ -444,6 +444,65 @@ export class XAccountController {
         log.info("XAccountController.initDB: database initialized");
     }
 
+    // Helper function to fetch tweets with media and URLs
+    private fetchTweetsWithMediaAndURLs(
+        whereClause: string,
+        params: (string | number)[]
+    ): XTweetItem[] {
+        const query = `
+            SELECT
+                t.tweetID, t.text, t.likeCount, t.retweetCount, t.createdAt,
+                tm.mediaType, tm.filename AS mediaFilename,
+                tu.expandedURL AS urlExpanded
+            FROM tweet t
+            LEFT JOIN tweet_media tm ON t.tweetID = tm.tweetID
+            LEFT JOIN tweet_url tu ON t.tweetID = tu.tweetID
+            WHERE ${whereClause}
+            ORDER BY t.createdAt ASC
+        `;
```

Comment:
I think we should ensure we use more joins for such situations throughout our codebase instead of needing to do multiple queries. 💯